package/src/components/MessageInput/hooks/useAudioController.tsx

```diff
@@ -38,13 +41,18 @@ export const useAudioController = () => {
   // For playback support in Expo CLI apps
   const soundRef = useRef<SoundReturnType | null>(null);
 
-  // Effect to stop the player when the component unmounts
-  useEffect(
-    () => () => {
+  // This effect controls the creation of an AudioClass instance during mounting
+  // and cleanup during unmounting (stopping both the player and a potential recording).
+  useEffect(() => {
+    if (AudioClass) {
+      // @ts-ignore
+      Audio = new AudioClass();
```

Comment:
Needed to switch to a class based approach for the Audio module because of the fact that the only reason `stopVoiceRecording()` was not pure was the fact that the Expo implementation depended on the recording object. We can easily keep track of this (since it's anyway only 2 methods for the Expo version) and dereference it that way.